apollo/src/graphql/storage-provider/storageProviderQueries.ts

```diff
@@ -27,4 +30,96 @@ builder.queryFields((t) => ({
             return storageProvider;
         },
     }),
+    listObjectsForModelVersion: t.field({
+        type: StorageProviderObjectConnection,
+        authScopes: {
+            loggedIn: true,
+        },
+        args: {
+            modelVersionId: t.arg.string({ required: true }),
+            // TODO: Putting defaultValue & required overrides defaultValue
+            first: t.arg({
+                type: 'Limit',
+                defaultValue: 10,
+                required: true,
+            }),
+            after: t.arg.string(),
+            prefix: t.arg.string(),
+        },
+        async resolve(_root, args, _ctx) {
+            const splitKey = (prefixLength: number, objectKey?: string): string[] => {
+                if (objectKey) {
+                    return objectKey.slice(prefixLength + 1).split('/');
+                }
+
+                return [];
+            };
+
+            const folderOrFile = (keyArr?: string[]): 'folder' | 'file' | undefined => {
```

Comment:
Do we benefit from differentiating?  What's the utility of this versus just grouping objects by greatest common subpath?

Reply:
I said that badly. Like, why not just create a tree structure internally that represents the nodes in a filepath? The concept of a folder isn't really meaningful beyond being a heuristic for better UX. We really only care about the blob names and can back into a pretty UI that splits things out into a tree structure that's familiar to an end user